lib/datasets.go

```diff
@@ -828,7 +828,15 @@ func (r *DatasetRequests) Remove(p *RemoveParams, res *RemoveResponse) error {
 	if p.Revision.Gen == dsref.AllGenerations {
 		// removing all revisions of a dataset must unlink it
 		if ref.FSIPath != "" {
-			if err := r.inst.fsi.Unlink(ref.FSIPath, ref.AliasString()); err == nil {
+			var err error
+			if p.Force {
+				err = r.inst.fsi.RemoveAll(ref.FSIPath, ref.AliasString())
+			} else if !p.KeepFiles {
+				err = r.inst.fsi.Remove(ref.FSIPath, ref.AliasString())
```

Comment:
This call is happening before `fsi.DeleteComponentFiles` below, which is what removes "body.csv", "structure.json", "meta.json" etc. That means in nearly all cases, the working directory will not be empty, and the call to `os.Remove` within `fsi.Remove` will fail.

There's already a call to `os.Remove` below on the next line (after `fsi.DeleteComponentFiles`). I think the logic of removing low-value files should be happening over there, instead.

Reply:
👍 Currently only works due to `os.RemoveAll` from above.